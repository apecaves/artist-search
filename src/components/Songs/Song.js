import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

function Song({ song, artistName }) {
  return <Link to={`/song/${artistName}/${song.title}`}>{song.title}</Link>;
}

Song.propTypes = {
  song: PropTypes.shape({
    title: PropTypes.string.isRequired
  }).isRequired,
  artistName: PropTypes.string.isRequired
};

export default Song;
