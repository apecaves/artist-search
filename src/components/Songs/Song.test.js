import React from 'react';
import { shallow } from 'enzyme';
import Song from './Song';

describe('Song component', () => {
  it('renders Song', () => {
    const wrapper = shallow(<Song song={{ title: '' }} artistName={''}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
  