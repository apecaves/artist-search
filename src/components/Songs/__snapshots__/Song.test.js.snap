// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Song component renders Song 1`] = `
<Link
  to="/song//"
/>
`;
